### DISCUSSION TITLE
How to check if a frontend.Variable is contained in a list, if the list is a secret witness.

### DISCUSSION THREAD

**Discussion by drakstik** [Consensys/gnark#1070] [Q&A] [answered]
I'm trying to assert if a frontend.Variable is part of a list of frontend.Variable, and maintain the list of frontend.Variable secret.

So I'm running something like:

```
type Circuit struct {
    Permissible_Values    `gnark:",secret"`
    Value                         `gnark:",public"`
}

func (circuit *Circuit) Define(api frontend.API) error {
    for i := 0; i < len(circuit.Permissible_Values); i++ {
		if circuit.Permissible_Values[i] == Value { // This if statement accesses Permissible_Values directly....
                    api.AssertIsEqual(circuit.Permissible_Values[i], Value)
                }
    }
}
```

Essentially, I want the circuit to fail if the Value in the circuit does not match any of the Permissible_Values, but I want the if statement itself to be a circuit assertion, since Permissible_Values should remain secret. I've looked at api.Select but I'm unsure how I'd use it in this context (or any other context, couldnt find many examples) and Issue #81 seems unresolved. 

I have looked around for similar discussions and honestly this is a pretty confusing topic! Any help is appreciated.

**Comment by ivokub** [comment, marked as answer]
You could use `isDifferent = api.Sub(1, api.IsZero(api.Sub(circuit.Permissible_Values[i], Value)))` and then multiply the results and assert that the product is 1.

**Reply by drakstik** [reply]
Great! You're awesome! Thanks for the fast response and all your work on this. I changed the title in hope that it helps others. I suspected I did not need a special function.
